### PR TITLE
ISPN-14247 Drop snakeyaml dependency from the CLI

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-14247

### CODE REVIEW

**Comment by pruivo** [comment]
any chance to use our reader here [1] as well? If so, we can remove snakeyaml from all our poms
[1] https://github.com/infinispan/infinispan/blob/b50dc0b8acec9ab33315bdab4c25cd7127fc1e84/server/rest/src/test/java/org/infinispan/rest/resources/CacheResourceV2Test.java#L1054

**Comment by tristantarrant** [comment]
Sure @pruivo , just updated iut

File: commons/all/src/test/java/org/infinispan/commons/configuration/io/yaml/YamlConfigurationReaderTest.java [hunk 1]
```diff
@@ -171,4 +174,21 @@ private void assertAttribute(YamlConfigurationReader yaml, String name, String v
...
+         Map<String, Object> identities = yaml.asMap();
+         assertEquals(1, identities.size());
+         List<Object> credentials = (List<Object>) identities.get("credentials");
+         assertNotNull(credentials);
+         assertEquals(3, credentials.size());
+         for (Object o : credentials) {
+            Map<String, Object> credential = (Map<String, Object>) o;
+            assertTrue(credential.containsKey("username"));
+            assertTrue(credential.containsKey("password"));
+         }
```
Comment (pruivo) [inline commons/all/src/test/java/org/infinispan/commons/configuration/io/yaml/YamlConfigurationReaderTest.java:191]: nitpick, for completeness
```suggestion
                  for (Object o : credentials) {
            Map<String, Object> credential = (Map<String, Object>) o;
            assertTrue(credential.containsKey("username"));
            assertTrue(credential.containsKey("password"));
            assertEquals("changeme", credential.get("password"));
            if ("my-user-1".equals(credential.get("username"))) {
               assertTrue(credential.containsKey("roles"));
               assertEquals(Collections.singletonList("admin"), credential.get("roles"));
            } else if ("my-user-2".equals(credential.get("username"))) {
               assertTrue(credential.containsKey("roles"));
               assertEquals(Collections.singletonList("monitor"), credential.get("roles"));
            } else {
               assertEquals("admin", credential.get("username"));
               assertFalse(credential.containsKey("roles"));
            }
         }
```

**Review by pruivo — COMMENTED** [review]
It is good to go. I just left a suggestion to make the test more complete.

**Review by pruivo — APPROVED** [review]

**Comment by pruivo** [comment]
an import is missing @tristantarrant (sorry!)

**Comment by tristantarrant** [comment]
@pruivo I guess we'll want a backport to 14.0.x

**Comment by pruivo** [comment]
merged! thanks @tristantarrant 



> @pruivo I guess we'll want a backport to 14.0.x

go for it!

**Pull request merged**